app/storage/slice/transaction.ts

```diff
@@ -86,7 +87,10 @@ export const createTransactionSlice: BackgroundStateCreator<
 
   markERC4337TransactionRejected: async (txId, data) => {
     await set(({ state, getERC4337TransactionType }) => {
-      const tx = state.pendingTransaction[txId]
+      const txIndex = state.pendingRequests.findIndex(
+        (r) => r.type === RequestType.Transaction && r.id === txId
+      )
```

Comment:
`txIndex` could be not found here, too.

Reply:
I decided to throw error when not found at this moment. And I also extracted a `StateActor` to collect common logic for storage state across app and background, see [f486087](https://github.com/pilagod/waallet-extension/pull/199/commits/f486087239d461c46fee0f7cce3f486f9e150019)

Reply:
Just found the typo after push 🤦‍♂️ [c53f13e](https://github.com/pilagod/waallet-extension/pull/199/commits/c53f13e994bdd889b79aa69164ef352e2a6680f2)

Reply:
> I decided to throw error when not found at this moment. And I also extracted a StateActor to collect common logic for storage state across app and background, see [f486087](https://github.com/pilagod/waallet-extension/pull/199/commits/f486087239d461c46fee0f7cce3f486f9e150019)

: I'm wondering what diff between `StateActor` and manager like `NetworkStorageManager` or `AccountStorageManager`) 🧐?

`StateActor` seems to dealing with transaction request  so far. Are there any actions going to be implemented?